File: cli/logger.go
package main

import (
    "log"
    "os"
)

func setupLogger(debug bool) *log.Logger {
    logger := log.New(os.Stdout, "Velvet: ", log.LstdFlags)
    if debug {
        logger.SetFlags(log.LstdFlags | log.Lshortfile)
    }
    return logger
}

File: core/src/logger.rs
pub struct Logger {
    debug: bool,
}

impl Logger {
    pub fn new(debug: bool) -> Self {
        Logger { debug }
    }

    pub fn log(&self, message: &str) {
        if self.debug {
            println!("[DEBUG] {}", message);
        } else {
            println!("{}", message);
        }
    }
}
